src/external_cli/arg_builder.rs

```diff
@@ -0,0 +1,208 @@
+/// A helper to make passing arguments to [`Command`](std::process::Command) more convenient.
+#[derive(Debug, Clone)]
+pub struct ArgBuilder(Vec<String>);
+
+impl ArgBuilder {
+    /// Create a new builder for command arguments.
+    pub const fn new() -> Self {
+        Self(Vec::new())
+    }
+
+    /// Add a singular argument.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_cli::external_cli::arg_builder::ArgBuilder;
+    /// ArgBuilder::new().arg("--release");
+    /// ```
+    pub fn arg<A>(mut self, arg: A) -> Self
+    where
+        A: Into<String>,
+    {
+        self.0.push(arg.into());
+        self
+    }
+
+    /// Add an argument with a value.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_cli::external_cli::arg_builder::ArgBuilder;
+    /// ArgBuilder::new().add_with_value("--bin", "bevy");
+    /// ```
+    pub fn add_with_value<A, V>(self, arg: A, value: V) -> Self
+    where
+        A: Into<String>,
+        V: Into<String>,
+    {
+        self.arg(arg).arg(value)
+    }
+
+    /// Add a boolean flag with the given name, if `value` is `true`.
```

Comment:
```suggestion
    /// Add a boolean flag with the given name only if `value` is `true`.
```

(Nit) I would like to clarify that setting `value` to `false` does not add anything to the arguments.